File: api-dot-net/Services/CodeExecution/Dto/CodeExecutionRequest.cs
namespace CjsApi.Services.CodeExecution.Dto
{

    public class CodeExecutionRequest
    {
        public string Language { get; set; } = string.Empty;
        public string SourceCode { get; set; } = string.Empty;
        public string? Input { get; set; }
        public int TimeLimitMs { get; set; } = 2000;
        public int MemoryLimitMb { get; set; } = 256;
        public List<TestCaseDto> TestCases { get; set; } = new();
    }
}

File: java-api/cjs-service/src/main/java/com/cjs/cjs_service/service/codeExecutionSerivce/CodeExecutionRequest.java
package com.cjs.cjs_service.service.codeExecutionSerivce;

import java.util.ArrayList;
import java.util.List;

import com.cjs.cjs_service.dto.TestCaseDto;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class CodeExecutionRequest {

    private String language = "";
    private String sourceCode = "";
    private String input; // nullable
    private int timeLimitMs = 2000;
    private int memoryLimitMb = 256;
    private List<TestCaseDto> testCases = new ArrayList<>();

    public CodeExecutionRequest(String language, String sourceCode, List<TestCaseDto> testCases) {
        this.language = language;
        this.sourceCode = sourceCode;
        this.testCases = testCases;
    }

}
